tests/system/utils/__init__.py

```diff
@@ -0,0 +1,25 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from airflow.utils.state import State
+
+
+def get_test_run(dag):
+    def test_run():
+        dag.clear(dag_run_state=State.NONE)
+        dag.run()
+
+    return test_run
```

Comment:
```suggestion
def get_test_run(dag):
    def test_run():
        dag.clear(dag_run_state=None)
        dag.run()

    return test_run
```

Reply:
Good simplification. It is needed, because we want to be able to change it's behaviour globally (for example during manual testing). You might want to - for example change log level or record additional information or start debugger. or start monitoring memory at the "run time" or do other things. This is purely to avoid having to do it manually in all tests that use it.

Reply:
Basically it's the same reason why we wanted to extract "list(dag.task) >> watcher()" to separate function (but it ain't so easy and there it isn't worth it indeed.

Reply:
I was thinking the same thing (like @uranusjr) yesterday when I was looking at the whole solution (after the proposed changes) from the higher perspective. Our goal was to have self-contained tests that are just DAGs, and by exporting more and more things outside, we lost part of this feature.

On the other hand, I agree with @potiuk that thanks to this 'export', we can easily change or update some behavior in one place. I haven't even thought about changing test configuration this way (log level etc.) but, yeah, this will be quite useful.
It's a trade-off between fully self-contained and easily editable. We can't have both at the same time.

Reply:
Yeah. This is really the balance between DRY and DAMP and while tests should be more DAMP than DRY than the actual code, some DRY-ness is also useful:

https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests